File: typescript-connector/src/models/Concept.ts
// Class from DFC Business Ontology: #Concept
import { SemanticObject } from "../core/SemanticObject.js";

export interface ConceptParams {
  certificateOf?: string;
  claimOf?: string;
  containerInformationOf?: string;
  geographicalOriginOf?: string;
  natureOriginOf?: string;
  partOriginOf?: string;
  typeOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
}
export class Concept extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Concept";
  }

  certificateOf?: string;
  claimOf?: string;
  containerInformationOf?: string;
  geographicalOriginOf?: string;
  natureOriginOf?: string;
  partOriginOf?: string;
  typeOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;

  constructor(
    semanticId: string,
    params?: ConceptParams,
  ) {
    super(semanticId);
    this.certificateOf = params?.certificateOf;
    this.claimOf = params?.claimOf;
    this.containerInformationOf = params?.containerInformationOf;
    this.geographicalOriginOf = params?.geographicalOriginOf;
    this.natureOriginOf = params?.natureOriginOf;
    this.partOriginOf = params?.partOriginOf;
    this.typeOf = params?.typeOf;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.semanticType = Concept.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:certificateOf", () => this.certificateOf);
    this.registerSemanticProperty("dfc-b:claimOf", () => this.claimOf);
    this.registerSemanticProperty("dfc-b:containerInformationOf", () => this.containerInformationOf);
    this.registerSemanticProperty("dfc-b:geographicalOriginOf", () => this.geographicalOriginOf);
    this.registerSemanticProperty("dfc-b:natureOriginOf", () => this.natureOriginOf);
    this.registerSemanticProperty("dfc-b:partOriginOf", () => this.partOriginOf);
    this.registerSemanticProperty("dfc-b:typeOf", () => this.typeOf);
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
  }
  static {
    SemanticObject.typeRegistry.set(Concept.SEMANTIC_TYPE, Concept);
  }
}

File: ruby-gem/lib/models/concept.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Concept
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class Concept < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:Concept".freeze

      # @return [String]
      attr_accessor :certificate_of

      # @return [String]
      attr_accessor :claim_of

      # @return [String]
      attr_accessor :container_information_of

      # @return [String]
      attr_accessor :geographical_origin_of

      # @return [String]
      attr_accessor :nature_origin_of

      # @return [String]
      attr_accessor :part_origin_of

      # @return [String]
      attr_accessor :type_of

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param certificateOf: nil, claimOf: nil, containerInformationOf: nil, geographicalOriginOf: nil, natureOriginOf: nil, partOriginOf: nil, typeOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, certificateOf: nil, claimOf: nil, containerInformationOf: nil, geographicalOriginOf: nil, natureOriginOf: nil, partOriginOf: nil, typeOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @certificate_of = certificateOf
        @claim_of = claimOf
        @container_information_of = containerInformationOf
        @geographical_origin_of = geographicalOriginOf
        @nature_origin_of = natureOriginOf
        @part_origin_of = partOriginOf
        @type_of = typeOf
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:Concept"
        registerSemanticProperty("dfc-b:certificateOf", &method("certificate_of")).valueSetter = method("certificate_of=")
        registerSemanticProperty("dfc-b:claimOf", &method("claim_of")).valueSetter = method("claim_of=")
        registerSemanticProperty("dfc-b:containerInformationOf", &method("container_information_of")).valueSetter = method("container_information_of=")
        registerSemanticProperty("dfc-b:geographicalOriginOf", &method("geographical_origin_of")).valueSetter = method("geographical_origin_of=")
        registerSemanticProperty("dfc-b:natureOriginOf", &method("nature_origin_of")).valueSetter = method("nature_origin_of=")
        registerSemanticProperty("dfc-b:partOriginOf", &method("part_origin_of")).valueSetter = method("part_origin_of=")
        registerSemanticProperty("dfc-b:typeOf", &method("type_of")).valueSetter = method("type_of=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
